dateparser/search/search.py

```diff
@@ -171,15 +172,21 @@ def __init__(self):
         self.available_language_map = self.loader.get_locale_map()
         self.search = _ExactLanguageSearch(self.loader)
 
-    def detect_language(self, text, languages):
-        if isinstance(languages, (list, tuple, Set)):
+    @apply_settings
+    def detect_language(self, text, languages, settings=None, detect_languages_function=None):
+        if detect_languages_function:
+            detected_languages = detect_languages_function(
+                text, confidence_threshold=settings.LANGUAGE_DETECTION_CONFIDENCE_THRESHOLD
+            )
+            detected_languages = map_languages(detected_languages) or settings.DEFAULT_LANGUAGES
+            return detected_languages[0] if detected_languages else []
```

Comment:
why do we return a list here as a fallback, is this right? In other code paths we do a different thing I think.

Reply:
Done 👍🏻